### PR TITLE
Check for InCombatLockDown() to decide which checkers can be used

### PR DESCRIPTION
**Pull request by InfusOnWoW**
In an hotfix CheckInteractDistance and IsItemInRange were made protected in combat.

Blizzard in their incompetence somehow managed to not release that in 10.2 as planned and didn't tell us about the plan to change it in 10.2. And didn't tell us in advance of the hotfix releasing, nor could they clarify whether the hotfix was intentionally for several hours after the hotfix was released.

### CODE REVIEW

File: LibRangeCheck-3.0/LibRangeCheck-3.0.lua [hunk 1]
```diff
@@ -631,7 +638,7 @@ local function addChecker(t, range, minRange, checker, info)
 end
 
 local function createCheckerList(spellList, itemList, interactList)
-  local res = {}
+  local res, resInCombat = {}, {}
```
Comment (InfusOnWoW) [inline LibRangeCheck-3.0/LibRangeCheck-3.0.lua:641]: We now maintain duplicated list for in combat and not in combat

File: LibRangeCheck-3.0/LibRangeCheck-3.0.lua [hunk 2]
```diff
@@ -977,103 +1011,157 @@ function lib:init(forced)
 end
 
 --- Return an iterator for checkers usable on friendly units as (**range**, **checker**) pairs.
-function lib:GetFriendCheckers()
-  return rcIterator(self.friendRC)
+-- @param inCombat if true, only checkers that can be used in combat ar returned
+function lib:GetFriendCheckers(inCombat)
```
Comment (InfusOnWoW) [inline LibRangeCheck-3.0/LibRangeCheck-3.0.lua:1015]: The addition of an parameter is technically not fully backwards compatible, as the caller could have used a bogus extra parameter. But I find that an acceptable thing to potentially break.

Also, here I choose to not change what GetFriendCheker etc returns. The other choice would have been to make inCombat the default.

So addons might need to adjust their call depending on what they want. The returned checker might not work in combat, but will not cause an lua error .

File: LibRangeCheck-3.0/LibRangeCheck-3.0.lua [hunk 3]
```diff
@@ -1506,19 +1594,21 @@ function lib:updateMeasurements()
...
-  for i, v in pairs(DefaultInteractList) do
-    local key = "interact: " .. i
-    local last = self.lastMeasurements[key]
-    local curr = CheckInteractDistance(unit, i) and true or false
-    if last == nil or last ~= curr then
-      if not t then
-        t = {}
-        t.x, t.y, t.stamp, t.states = x, y, now, {}
-        self.measurements[now] = t
+  if not InCombatLockdown() then
```
Comment (InfusOnWoW) [inline LibRangeCheck-3.0/LibRangeCheck-3.0.lua:1597]: This is debug code, that isn't too important.

File: LibRangeCheck-3.0/LibRangeCheck-3.0.lua (hunk 2 above)
Comment (Stanzilla) [reply LibRangeCheck-3.0/LibRangeCheck-3.0.lua:1022]: I think that's fine.  @kodewdle does ElvUI need to adjust for this?

Comment (AcidWeb) [reply LibRangeCheck-3.0/LibRangeCheck-3.0.lua:1022]: ElvUI don't use this call.

**Review by mrbuds — APPROVED** [review]

**Pull request merged**